new_server.py

```diff
@@ -0,0 +1,28 @@
+from gevent.monkey import patch_all
+from gevent.server import StreamServer
+from server import parse_request
+
+
+def new_server(socket, address):
+    buffsize = 16
+    try:
+        msg = ""
+        complete = False
+        while not complete:
+            line = socket.recv(buffsize)
+            if line:
+                msg += line
+            else:
+                socket.sendall(parse_request(msg))
+                socket.close()
+                complete = True
```

Comment:
You are sending back a reply while you still may be receiving pieces of the request.  Break out of the loop when you are done accumulating the request and then handle it outside the loop.